src/lib.rs

```diff
@@ -94,81 +97,85 @@ impl nng_ctx {
     /// Converts value returned by [nng_stat_type](https://nanomsg.github.io/nng/man/v1.1.0/nng_stat_type.3) into `nng_stat_type_enum`.
-    pub fn try_from(value: i32) -> Result<Self, TryFromIntError> {
+    pub fn try_convert_from(value: i32) -> Option<Self> {
         use crate::nng_stat_type_enum::*;
         match value {
-            value if value == NNG_STAT_SCOPE as i32 => Ok(NNG_STAT_SCOPE),
-            value if value == NNG_STAT_LEVEL as i32 => Ok(NNG_STAT_LEVEL),
-            value if value == NNG_STAT_COUNTER as i32 => Ok(NNG_STAT_COUNTER),
-            value if value == NNG_STAT_STRING as i32 => Ok(NNG_STAT_STRING),
-            value if value == NNG_STAT_BOOLEAN as i32 => Ok(NNG_STAT_BOOLEAN),
-            value if value == NNG_STAT_ID as i32 => Ok(NNG_STAT_ID),
-            _ => Err(TryFromIntError),
+            value if value == NNG_STAT_SCOPE as i32 => Some(NNG_STAT_SCOPE),
+            value if value == NNG_STAT_LEVEL as i32 => Some(NNG_STAT_LEVEL),
+            value if value == NNG_STAT_COUNTER as i32 => Some(NNG_STAT_COUNTER),
+            value if value == NNG_STAT_STRING as i32 => Some(NNG_STAT_STRING),
+            value if value == NNG_STAT_BOOLEAN as i32 => Some(NNG_STAT_BOOLEAN),
+            value if value == NNG_STAT_ID as i32 => Some(NNG_STAT_ID),
+            _ => None,
         }
     }
 }
 
+#[cfg(try_from)]
+impl TryFrom<i32> for nng_stat_type_enum {
+    type Error = EnumFromIntError;
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        nng_stat_type_enum::try_convert_from(value).ok_or(EnumFromIntError(value))
+    }
+}
+
 impl nng_unit_enum {
-    // TODO: 1.33/1.34 replace this with TryFrom once stabilized:
-    // https://doc.rust-lang.org/std/convert/trait.TryFrom.html
     /// Converts value returned by [nng_stat_unit](https://nanomsg.github.io/nng/man/v1.1.0/nng_stat_unit.3) into `nng_unit_enum`.
-    pub fn try_from(value: i32) -> Result<Self, TryFromIntError> {
+    pub fn try_convert_from(value: i32) -> Option<Self> {
         use crate::nng_unit_enum::*;
         match value {
-            value if value == NNG_UNIT_NONE as i32 => Ok(NNG_UNIT_NONE),
-            value if value == NNG_UNIT_BYTES as i32 => Ok(NNG_UNIT_BYTES),
-            value if value == NNG_UNIT_MESSAGES as i32 => Ok(NNG_UNIT_MESSAGES),
-            value if value == NNG_UNIT_MILLIS as i32 => Ok(NNG_UNIT_MILLIS),
-            value if value == NNG_UNIT_EVENTS as i32 => Ok(NNG_UNIT_EVENTS),
-            _ => Err(TryFromIntError),
+            value if value == NNG_UNIT_NONE as i32 => Some(NNG_UNIT_NONE),
+            value if value == NNG_UNIT_BYTES as i32 => Some(NNG_UNIT_BYTES),
+            value if value == NNG_UNIT_MESSAGES as i32 => Some(NNG_UNIT_MESSAGES),
+            value if value == NNG_UNIT_MILLIS as i32 => Some(NNG_UNIT_MILLIS),
+            value if value == NNG_UNIT_EVENTS as i32 => Some(NNG_UNIT_EVENTS),
+            _ => None,
         }
     }
 }
 
-impl nng_pipe_ev {
```

Comment:
This enum doesn't match the others with `try_convert_from` and ` TryFrom` implementation. It's small enough to not really matter, though.

Reply:
If I remember correctly, those were just a temporary solution before `TryFrom` was stabilized, and `nng_pipe_ev` didn't exist then.  Once we no longer care about Rust <1.34 we can just remove `try_convert_from`.  Doesn't make sense to make them deprecated since there isn't a replacement for earlier rust.